test/SIL/ownership-verifier/interior_pointer.sil

```diff
@@ -125,19 +125,18 @@ bb0(%0 : @owned $Builtin.NativeObject):
 
 // CHECK-LABEL: Error#: 0. Begin Error in Function: 'recursive_interior_pointer_error'
 // CHECK-NEXT: Found outside of lifetime use?!
-// CHECK-NEXT: Value:   %2 = begin_borrow %0 : $KlassUser               // users: %5, %3
-// CHECK-NEXT: Consuming User:   end_borrow %2 : $KlassUser                      // id: %5
-// CHECK-NEXT: Non Consuming User:   %7 = load [copy] %4 : $*Klass                   // user: %8
+// CHECK-NEXT: Value:   %2 = begin_borrow %0 : $KlassUser
+// CHECK-NEXT: Consuming User:   end_borrow %2 : $KlassUser
+// CHECK-NEXT: Non Consuming User:   %6 = load [copy] %3 : $*Klass
 // CHECK-NEXT: Block: bb0
 // CHECK: Error#: 0. End Error in Function: 'recursive_interior_pointer_error'
 sil [ossa] @recursive_interior_pointer_error : $@convention(thin) (@owned KlassUser, @guaranteed Klass) -> @owned Klass {
 bb0(%0 : @owned $KlassUser, %0a : @guaranteed $Klass):
   %1 = begin_borrow %0 : $KlassUser
   %2 = ref_tail_addr %1 : $KlassUser, $Klass
-  %result = store_borrow %0a to %2 : $*Klass
```

Comment:
This was testing that we are able to flag errors on store_borrow. This makes me a bit nervous about restricting store_borrow only to alloc_stack. I think this test case IIRC was from real code that we are emitting today. Now that I think further, one could imagine eliminating a temporary rvalue causing a store_borrow to be moved off of an alloc_stack.

Reply:
Talked with eeckstein offline. This test is making sure that the use of %result is treated as a liveness requirement of %0. So by removing the store_borrow the test isn't testing anything anymore. Can you delete it?

Also for anyone listening in, we actually do not need to test this anymore since this PR makes it so that store_borrow's destination is always an alloc_stack meaning it can never store to an interior pointer destination.